src/bespokelabs/curator/prompter/prompter.py

```diff
@@ -286,25 +288,157 @@ def _completions(
         return dataset
 
 
-def _get_function_hash(func) -> str:
-    """Get a hash of a function's source code."""
-    if func is None:
-        return xxh64("").hexdigest()
+class PathIndependentPickler:
```

Comment:
why can't we just inherit from dill Pickler like HF pickler? seems like you're remimplementing a lot of details from dill Pickler.

https://github.com/huggingface/datasets/blob/2049c00921c59cdeb835137a1c49639cf175af07/src/datasets/utils/_dill.py#L27

specifically for codeobject pickling look at https://github.com/huggingface/datasets/blob/2049c00921c59cdeb835137a1c49639cf175af07/src/datasets/utils/_dill.py#L223C34-L224C1